examples/webgl_loader_gainmap.html

```diff
@@ -0,0 +1,326 @@
+			Gain map images converted from hdr with <a href="https://gainmap-creator.mono-grid.com/" target="_blank" rel="noopener">Gain map converter</a>. <br />
+			See external <a href="https://github.com/MONOGRID/gainmap-js" target="_blank" rel="noopener">gainmap-js</a> for more information on how to use and create gain map images.
+		</div>
+
+		<div id="lbl_left" class="lbl"></div>
+
+		<script type="importmap">
+			{
+				"imports": {
+					"three": "../build/three.module.js",
+					"three/addons/": "./jsm/",
+					"@monogrid/gainmap-js": "https://unpkg.com/@monogrid/gainmap-js@2.0.5/dist/decode.js"
+				}
+			}
+		</script>
+
+		<script type="module">
+
+			import * as THREE from 'three';
+
+			import Stats from 'three/addons/libs/stats.module.js';
+
+			import { GUI } from 'three/addons/libs/lil-gui.module.min.js';
+			import { OrbitControls } from 'three/addons/controls/OrbitControls.js';
+			import { RGBELoader } from 'three/addons/loaders/RGBELoader.js';
+
+			import { GainMapLoader, HDRJPGLoader } from '@monogrid/gainmap-js';
+
+			const params = {
+				envMap: 'HDR JPG',
+				roughness: 0.0,
+				metalness: 0.0,
+				exposure: 1.0,
+				debug: false
+			};
+
+			let container, stats;
+			let camera, scene, renderer, controls;
+			let torusMesh, planeMesh;
+			let hdrJpg, hdrJpgPMREMRenderTarget, hdrJpgEquirectangularMap;
+			let gainMap, gainMapPMREMRenderTarget, gainMapBackground;
+			let hdrPMREMRenderTarget, hdrEquirectangularMap;
+
+			
+			const fileSizes = {};
+			const resolutions = {};
+
+			init();
+			animate();
+
+			function init() {
+
+				const lbl = document.getElementById( 'lbl_left' );
+
+				container = document.createElement( 'div' );
+				document.body.appendChild( container );
+
+				camera = new THREE.PerspectiveCamera( 40, window.innerWidth / window.innerHeight, 1, 1000 );
+				camera.position.set( 0, 0, 120 );
+
+				scene = new THREE.Scene();
+				scene.background = new THREE.Color( 0x000000 );
+
+				renderer = new THREE.WebGLRenderer();
+				renderer.toneMapping = THREE.ACESFilmicToneMapping;
+
+				//
+
+				let geometry = new THREE.TorusKnotGeometry( 18, 8, 150, 20 );
+				// let geometry = new THREE.SphereGeometry( 26, 64, 32 );
+				let material = new THREE.MeshStandardMaterial( {
+					color: 0xffffff,
+					metalness: params.metalness,
+					roughness: params.roughness
+				} );
+
+				torusMesh = new THREE.Mesh( geometry, material );
+				scene.add( torusMesh );
+
+
+				geometry = new THREE.PlaneGeometry( 200, 200 );
+				material = new THREE.MeshBasicMaterial();
+
+				planeMesh = new THREE.Mesh( geometry, material );
+				planeMesh.position.y = - 50;
+				planeMesh.rotation.x = - Math.PI * 0.5;
+				scene.add( planeMesh );
+
+
+				const pmremGenerator = new THREE.PMREMGenerator( renderer );
+				pmremGenerator.compileEquirectangularShader();
+
+				THREE.DefaultLoadingManager.onLoad = function ( ) {
+
+					pmremGenerator.dispose();
```

Comment:
Is this doing anything? It does not appear to cause the generator to be disposed.

Reply:
I'm not sure, I wrote my example starting from [this one](https://github.com/mrdoob/three.js/blob/4567ecee2aa19584533260d331fd80d8100b2240/examples/webgl_materials_envmaps_hdr.html#L91) and I assumed it disposed the pmremgenerator, if it's not doing it then I can remove it

Reply:
You can type `renderer.info` into the console.

Reply:
okay, my loaders need to handle `manager.itemStart` & `manager.itemEnd` in order to honor the end user's `DefaultLoadingManager.onLoad`  (which needs to be called after all loading has been completed).

I had delegated signaling of these events to their internally used `FileLoader`s but I actually need to wait until after the gainMap is rendered and returned to the end user before marking `manager.itemEnd`.

New version with the fix is coming up

Reply:
this is now fixed in version `2.0.6`, `THREE.DefaultLoadingManager.onLoad` is called after all callbacks have executed so  `pmremGenerator` can properly be disposed